### PR TITLE
Feature/proxy sources client refactor

### PR DESCRIPTION
**Pull request by git-alexbondarenko**

### CODE REVIEW

File: src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java [hunk 1]
```diff
@@ -1,11 +1,48 @@
...
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
 
 public class ProxySourcesClientImpl implements ProxySourcesClient {
+    private final ProxyValidator proxyValidator;
+    private static final Logger log = LoggerFactory.getLogger(ProxySourcesClientImpl.class);
```
Comment (Sokol3783) [inline src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java:21]: As I remember static constants must be on first place

**Review by hromovyi-kostiantyn — APPROVED** [review]

File: src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java [hunk 2]
```diff
@@ -1,11 +1,48 @@
...
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
 
 public class ProxySourcesClientImpl implements ProxySourcesClient {
```
Comment (OlexTul) [inline src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java:19]: 1. Reading ProxyNetworkConfig, ProxyCredentials from different files: proxy-network.json, proxy-credential.json:
image

2. As a result, it is necessary to ensure that the read ProxyNetworkConfig and ProxyCredentials are correctly combined into the ProxyConfigHolder (the problem may be a different number of elements in the proxy-network.json, proxy-credential.json files).

3. It is also necessary to simulate the situation of random data arrival for implementation in ParralelFlowExecutorService.

In my opinion, pull-request #43 is more suitable for these tasks.

**Review by OlexTul — CHANGES_REQUESTED** [review]

File: src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java (hunk 2 above)
Comment (git-alexbondarenko) [reply src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java:19]: 1. That's one way of solving this task. In my opinion, not the best one

2. That's the main reason why I propose to combine these files into one. Either one has no use without other - ProxyNetworkConfig will not be valid without proper credentials; credentials will have no use without ProxyNetworkConfig

3. I suppose, ParralelFlowExecutorService consumes ProxyConfigHolder, so I don't get your point. Also, proxies come through validation before being added to list

Comment (OlexTul) [reply src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java:19]: I just noticed the annul contract.

**Pull request closed without merging**